src/nodes_common.c

```diff
@@ -568,8 +568,12 @@ create_append_plan_common(PlannerInfo *root, RelOptInfo *rel,
 		forboth (lc1, rpath->cpath.custom_paths, lc2, custom_plans)
 		{
 			Plan		   *child_plan = (Plan *) lfirst(lc2);
-			RelOptInfo 	   *child_rel = ((Path *) lfirst(lc1))->parent;
+			RelOptInfo	   *child_rel = ((Path *) lfirst(lc1))->parent;
+#if PG_VERSION_NUM >= 110000
+			AppendRelInfo *appinfo = root->append_rel_array[child_rel->relid];
```

Comment:
NOTE: this is closely related to `setup_append_rel_array`.

Reply:
Well, except for access in nodes_common.c all others are just iterations over the whole list, so not sure we need to change something here.